bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/model/SensiboSky.java

```diff
@@ -12,42 +12,47 @@
  */
 package org.openhab.binding.sensibo.internal.model;
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import javax.measure.Unit;
 import javax.measure.quantity.Temperature;
 
 import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.core.library.unit.ImperialUnits;
 import org.eclipse.smarthome.core.library.unit.SIUnits;
-import org.openhab.binding.sensibo.internal.dto.poddetails.Measurement;
 import org.openhab.binding.sensibo.internal.dto.poddetails.ModeCapability;
 import org.openhab.binding.sensibo.internal.dto.poddetails.PodDetails;
-import org.openhab.binding.sensibo.internal.dto.poddetails.Schedule;
 
 /**
  * The {@link SensiboSky} represents a Sensibo Sky unit
  *
  * @author Arne Seime - Initial contribution
  */
+@NonNullByDefault
 public class SensiboSky extends Pod {
 
     private final String macAddress;
     private final String firmwareVersion;
     private final String firmwareType;
     private final String serialNumber;
-    private Unit<Temperature> temperatureUnit;
-    private final String originalTemperatureUnit;
+    private Unit<Temperature> temperatureUnit = SIUnits.CELSIUS;
+    private String originalTemperatureUnit = "C";
     private final String productModel;
-    private Boolean smartMode;
-    private AcState acState = null;
-    private Double temperature = null;
-    private Double humidity = null;
+    private Optional<AcState> acState = Optional.empty();;
+    private Double temperature;
+    private Double humidity;
     private boolean alive = false;
     private final String roomName;
-    private Schedule[] schedules;
+    private Schedule[] schedules = new Schedule[0];
+    private final Map<String, ModeCapability> remoteCapabilities;
+    private Optional<Timer> timer = Optional.empty();
```

Comment:
You could also do `private @Nullable timer` and use `return Optional.ofNullable(timer);` in the getter. The null check in the constructor could be remloved then. But ok for me the way it is implemented now.

Reply:
Leaving as is for now